tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -0,0 +1,1050 @@
+                    () -> "Timeout waiting for partition metadata propagating to brokers for underMinIsrTopic topic",
+                    org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS, 100L);
+            }
+            String output = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--describe", "--under-min-isr-partitions"));
+            String[] rows = output.split("\n");
+            assertTrue(rows[0].startsWith(String.format("Topic: %s", underMinIsrTopic)));
+            assertTrue(rows[1].startsWith(String.format("\tTopic: %s", offlineTopic)));
+            assertEquals(2, rows.length);
+        } finally {
+            restartDeadBrokers(false);
+        }
+    }
+
+    @ParameterizedTest(name = ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_NAME)
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testDescribeReportOverriddenConfigs(String quorum) throws Exception {
+        String config = "file.delete.delay.ms=1000";
+        createAndWaitTopic(buildTopicCommandOptionsWithBootstrap("--create", "--partitions", "2",
+            "--replication-factor", "2", "--topic", testTopicName, "--config", config));
+        String output = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--describe"));
+        assertTrue(output.contains(config), String.format("Describe output should have contained %s", config));
+    }
+
+    @ParameterizedTest(name = ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_NAME)
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testDescribeAndListTopicsWithoutInternalTopics(String quorum) throws Exception {
+        createAndWaitTopic(
+            buildTopicCommandOptionsWithBootstrap("--create", "--partitions", "1", "--replication-factor", "1", "--topic", testTopicName));
+        // create a internal topic
+        createAndWaitTopic(
+            buildTopicCommandOptionsWithBootstrap("--create", "--partitions", "1", "--replication-factor", "1", "--topic", Topic.GROUP_METADATA_TOPIC_NAME));
+
+        // test describe
+        String output = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--describe", "--describe", "--exclude-internal"));
+        assertTrue(output.contains(testTopicName),
+            String.format("Output should have contained %s", testTopicName));
+        assertFalse(output.contains(Topic.GROUP_METADATA_TOPIC_NAME));
+
+        // test list
+        output = captureListTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--list", "--exclude-internal"));
+        assertTrue(output.contains(testTopicName));
+        assertFalse(output.contains(Topic.GROUP_METADATA_TOPIC_NAME));
+    }
+
+    @ParameterizedTest(name = ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_NAME)
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testDescribeDoesNotFailWhenListingReassignmentIsUnauthorized(String quorum) throws ExecutionException, InterruptedException {
```

Comment:
We don't seem to have logic to have the `listPartitionReassignments()` call fail with `ClusterAuthorizationException`

Reply:
good catch I updated the test.